working-docs/transient.md

```diff
@@ -0,0 +1,379 @@
+
+#### Description
+
+Microarchitectural predictors may allow incorrect operations (or correct
+operations with incorrect data) to execute transiently after a
+misprediction.
+
+#### Extended Description
+
+Many commodity processors use microarchitectural predictors (for
+example, branch predictors) to improve performance. Operations that
+execute after a misprediction are not committed to the processor’s
+architectural state and their results are discarded. However, these
+transient operations can affect observable microarchitectural state in a
+manner that could allow an attacker to infer program data.
+
+Some commodity processors may provide features that prevent
+microarchitectural predictors from allowing program data to become
+inferable by an attacker. For example, a processor may allow software to
+opt-in to temporarily disable a microarchitectural predictor. Many
+instruction set architectures also provide serialization instructions
+that can be used by software to prevent predictions made prior to the
+serialization instruction from causing transient execution after the
+instruction. Some modern compilers can be directed to automatically
+instrument their output to mitigate the effects of transient execution
+caused by microarchitectural predictors, especially branch predictors.
+
+Developers of sandbox or managed runtime software should exercise
+caution when allowing multiple users to run code within the same
+hardware domain. For example, one user’s code may be able to train a
+microarchitectural predictor in a manner that allows it to transiently
+read another user’s data. As a second example, one user’s code may be
+able to train a microarchitectural predictor in a manner that influences
+transient execution when another user runs code, potentially causing
+that user’s data to become inferable. In most cases, these weaknesses
+can be mitigated by isolating each user’s code and data in separate
+hardware domains.
+
+Author’s Notes
+--------------------------------
+
+  - When writing these descriptions, I have tried as much as possible to
+    avoid introducing new terms, or terms that are specific to Intel’s
+    literature/documentation.
+    
+      - Intel often uses “malicious adversary,” but here I have used
+        “attacker” to align with the [CWE
+        glossary](https://cwe.mitre.org/documents/glossary/).
+    
+      - “Transient,” “transient execution,” “transient operations,” etc.
+        do not exist in the CWE glossary but have been used in other
+        CWEs. I had initially drafted these descriptions without using
+        the word “transient,” but this often led to unappealing
+        verbosity—I had to spell out “operations that execute but do not
+        commit to architectural state” many times. Perhaps MITRE should
+        consider adding “transient” to its CWE glossary.
+    
+      - I make liberal use of the term “processor event,” which is
```

Comment:
Gananand: Should we perhaps make a list of all processor events?

Reply:
Steven: If "all" events is too large, it would probably be good to list a few common/popular ones.